src/v/cloud_storage/types.cc

```diff
@@ -450,23 +450,30 @@ ss::future<configuration> configuration::get_abs_config() {
     const auto container = get_value_or_throw(
       config::shard_local_cfg().cloud_storage_azure_container,
       "cloud_storage_azure_container");
-    const auto shared_key = cloud_roles::private_key_str{get_value_or_throw(
-      config::shard_local_cfg().cloud_storage_azure_shared_key,
-      "cloud_storage_azure_shared_key")};
+    const auto shared_key =
+      []() -> std::optional<cloud_roles::private_key_str> {
+        auto opt
+          = config::shard_local_cfg().cloud_storage_azure_shared_key.value();
+        if (opt.has_value()) {
+            return cloud_roles::private_key_str(opt.value());
+        }
+        return std::nullopt;
+    }();
 
     const auto cloud_credentials_source
       = config::shard_local_cfg().cloud_storage_credentials_source.value();
-    if (
-      cloud_credentials_source
-      != model::cloud_credentials_source::config_file) {
+    using enum model::cloud_credentials_source;
+    if (!(cloud_credentials_source == config_file
+          || cloud_credentials_source == azure_aks_oidc_federation
+          || cloud_credentials_source == azure_vm_instance_metadata)) {
```

Comment:
this change looks unrelated to its commit

> cloud_storage/types: update configuration::get_abs_config
shared_key is not present if the cluster is using managed identities.
nonetheless, abs_config is used also there